x/evm/keeper/grpc_query_test.go

```diff
@@ -447,38 +467,518 @@ func (s *KeeperSuite) TestQueryParams() {
+	// TX 1: Deploy ERC-20 contract
+	contractData := evmtest.SmartContract_FunToken.Load(s.T())
+	nonce := deps.StateDB().GetNonce(deps.Sender.EthAddr)
+	txArgs := evm.JsonTxArgs{
+		From:  &deps.Sender.EthAddr,
+		Nonce: (*hexutil.Uint64)(&nonce),
+		Data:  (*hexutil.Bytes)(&contractData.Bytecode),
+	}
+	ethTxMsg, err := GenerateAndSignEthTxMsg(txArgs, deps)
+	s.NoError(err)
 
-	jsonTxArgs, err := json.Marshal(&evm.JsonTxArgs{
-		From: &deps.Sender.EthAddr,
-		Data: (*hexutil.Bytes)(&bytecode),
-	})
+	resp, err := deps.Chain.EvmKeeper.EthereumTx(deps.GoCtx(), ethTxMsg)
 	s.Require().NoError(err)
+	s.Require().Empty(resp.VmError)
 
-	_, err = deps.Chain.EvmKeeper.EstimateGas(deps.GoCtx(), &evm.EthCallRequest{
-		Args:            jsonTxArgs,
-		GasCap:          srvconfig.DefaultGasCap,
-		ProposerAddress: []byte{},
-		ChainId:         deps.Chain.EvmKeeper.EthChainID(deps.Ctx).Int64(),
-	})
+	// Contract address is deterministic
+	contractAddress := crypto.CreateAddress(deps.Sender.EthAddr, nonce)
+	deps.Chain.Commit()
+	predecessors := []*evm.MsgEthereumTx{
+		ethTxMsg,
+	}
+
+	// TX 2: execute ERC-20 contract transfer
+	input, err := contractData.ABI.Pack(
+		"transfer", GenerateEthAddress(), new(big.Int).SetUint64(1000),
+	)
+	s.NoError(err)
+	nonce = deps.StateDB().GetNonce(deps.Sender.EthAddr)
+	txArgs = evm.JsonTxArgs{
+		From:  &deps.Sender.EthAddr,
+		To:    &contractAddress,
+		Nonce: (*hexutil.Uint64)(&nonce),
+		Data:  (*hexutil.Bytes)(&input),
+	}
+	ethTxMsg, err = GenerateAndSignEthTxMsg(txArgs, deps)
+	s.NoError(err)
+
+	resp, err = deps.Chain.EvmKeeper.EthereumTx(deps.GoCtx(), ethTxMsg)
 	s.Require().NoError(err)
+	s.Require().Empty(resp.VmError)
 
-	_, err = deps.Chain.EvmKeeper.EthCall(deps.GoCtx(), &evm.EthCallRequest{
-		Args:            jsonTxArgs,
+	return ethTxMsg, predecessors
+}
+
+// GenerateAndSignEthTxMsg estimates gas, sets gas limit and sings the tx
+func GenerateAndSignEthTxMsg(txArgs evm.JsonTxArgs, deps *evmtest.TestDeps) (*evm.MsgEthereumTx, error) {
+	estimateArgs, err := json.Marshal(&txArgs)
+	if err != nil {
+		return nil, err
+	}
+	res, err := deps.Chain.EvmKeeper.EstimateGas(deps.GoCtx(), &evm.EthCallRequest{
+		Args:            estimateArgs,
 		GasCap:          srvconfig.DefaultGasCap,
 		ProposerAddress: []byte{},
 		ChainId:         deps.Chain.EvmKeeper.EthChainID(deps.Ctx).Int64(),
 	})
-	s.Require().NoError(err)
+	if err != nil {
+		return nil, err
+	}
+	txArgs.Gas = (*hexutil.Uint64)(&res.Gas)
+
+	txMsg := txArgs.ToTransaction()
+	gethSigner := deps.Sender.GethSigner(deps.Chain.EvmKeeper.EthChainID(deps.Ctx))
+	keyringSigner := deps.Sender.KeyringSigner
+	return txMsg, txMsg.Sign(gethSigner, keyringSigner)
+}
+
+func GenerateEthAddress() gethcommon.Address {
+	privateKey, _ := crypto.GenerateKey()
+	publicKey := privateKey.Public()
+	publicKeyECDSA, _ := publicKey.(*ecdsa.PublicKey)
+	return crypto.PubkeyToAddress(*publicKeyECDSA)
```

Comment:
This should be replaced with the equivalent util fn from `evmtest` if possible. It looks almost the same as `evmtest.NewEthAccInfo` from eth.go. 

If these functions are not the same, can you add it to `evmtest` and put a function comment on the difference?